plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergFileWriter.java

```diff
@@ -22,6 +24,15 @@
 public interface IcebergFileWriter
         extends FileWriter
 {
+    FileFormat getFileFormat();
+
+    String location();
+
+    default List<String> rewrittenDeleteFiles()
+    {
+        return ImmutableList.of();
+    }
```

Comment:
This method feels somehow artificial for the generic `IcebergFileWriter`

Reply:
I see it used in IcebergMetadat#finishWrite for `PUFFIN` only from within the `CommitTaskData`

I don't have right now a solution to suggest, but I feel we can do better also with `CommitTaskData`.
It currently contains a set of seemingly fields.

Did we explore the avenue of having `DataCommitTaskData` & `PositionDeleteCommitTaskData` (& `DeletionVectorCommitTaskData`) ?